### PR TITLE
added starfix to work without default config

### PR DESCRIPTION
**Pull request by zedbeit**
Made starfix to work without config on windows and provide implementations for two IDEs, VScode and IntelliJ.

### CODE REVIEW

File: cli/src/main/java/dev/starfix/Starfix.java [hunk 1]
```diff
@@ -134,6 +152,44 @@ public static File getConfigFile() {
...
+            configFile.createNewFile();
+            
+            if(isWindows()){// check if Windows OS
+                if(path_env.contains("Microsoft VS Code")){ // If PATH has VScode
+                    ide = "code.cmd";
+                    writeToYAMLFile(ide, clone_path,configFile);
+                } else if(path_env.contains("IntelliJ IDEA")){ // If PATH has IntelliJ
+                    ide = "idea64.exe";
+                    writeToYAMLFile(ide, clone_path,configFile);
+                }
```
Comment (maxandersen) [inline cli/src/main/java/dev/starfix/Starfix.java:171]: We shouldn't have to write the default config to disk. Then it by definition is no longer a default.

File: cli/src/main/java/dev/starfix/Starfix.java [hunk 2]
```diff
@@ -55,6 +55,14 @@ public int config() throws Exception {
 
     @Command(name = "clone")
     public int cloneCmd(@Parameters(index = "0") String url) {
+        String userHome =  System.getProperty("user.home"); // Get User Home Directory: /home/user_name
```
[the remark this answers is not in the document]
Comment (zedbeit) [reply cli/src/main/java/dev/starfix/Starfix.java]: I noticed that the getConfigFile creates ".config" directory, which won't be useful, that's why i didn't use it.

File: cli/src/main/java/dev/starfix/Starfix.java [hunk 3]
```diff
@@ -116,25 +127,64 @@ public static boolean isBlob(String url){
...
+                }
+            }
+            // check if Linux OS
+            // if(isLinux()){}
+            
+            // check if Mac OS
+            // if(isMac()){}
 
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (maxandersen) [inline cli/src/main/java/dev/starfix/Starfix.java:184]: we shuoldn't just be printing stacktrace. is this try/catch even needed?

File: cli/src/main/java/dev/starfix/Starfix.java [hunk 4]
```diff
@@ -55,6 +55,17 @@ public int config() throws Exception {
...
     @Command(name = "clone")
     public int cloneCmd(@Parameters(index = "0") String url) {
+        File configFilePath = getConfigFilePath(); // Get path for config file 
+
+        try {
+            if (!configFilePath.exists()) {// Check if config file exist
+                defaultConfig();
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
```
Comment (maxandersen) [inline cli/src/main/java/dev/starfix/Starfix.java:65]: should not print stakctraces raw to the user. and what do we expect that can throw an exception here?

File: cli/src/main/java/dev/starfix/Starfix.java [hunk 5]
```diff
@@ -116,25 +127,64 @@ public static boolean isBlob(String url){
...
+            if(!configFilePath.getParentFile().mkdirs()){// Create parent dir if not exist
+                throw new IllegalStateException("Cannot create .config directory: " + configFilePath.getParentFile().getAbsolutePath());
             }
         }
-        return new File(userHome + "/.config/starfix.yaml");
+        return configFilePath;
 
     }
+    
+    // Function to load default config
```
Comment (maxandersen) [inline cli/src/main/java/dev/starfix/Starfix.java:164]: it does not load anything ,right ? it just setups default configuration.

File: cli/src/main/java/dev/starfix/Starfix.java [hunk 6]
```diff
@@ -116,25 +127,64 @@ public static boolean isBlob(String url){
...
+                throw new IllegalStateException("Cannot create .config directory: " + configFilePath.getParentFile().getAbsolutePath());
             }
         }
-        return new File(userHome + "/.config/starfix.yaml");
+        return configFilePath;
 
     }
+    
+    // Function to load default config 
+    public void defaultConfig() {
```
Comment (maxandersen) [inline cli/src/main/java/dev/starfix/Starfix.java:165]: no need for this to be public.

**Review by maxandersen — COMMENTED** [review]
general approach is sound enough - avoids storing file just to get a default config. 


but a few issues around exception handling we should fix.

File: cli/src/main/java/dev/starfix/Starfix.java (hunk 4 above)
Comment (zedbeit) [reply cli/src/main/java/dev/starfix/Starfix.java]: I thought the code block might throw an exception, but I've fixed that now.

File: cli/src/main/java/dev/starfix/Starfix.java (hunk 5 above)
Comment (zedbeit) [reply cli/src/main/java/dev/starfix/Starfix.java]: Yeah it just sets up default config.

**Review by maxandersen — APPROVED** [review]

**Pull request merged**